h2/src/main/org/h2/schema/Schema.java

```diff
@@ -672,10 +672,9 @@ public Table getTableOrViewByName(String name) {
     public void remove(SchemaObject obj) {
         String objName = obj.getName();
         Map<String, SchemaObject> map = getMap(obj.getType());
-        if (SysProperties.CHECK && !map.containsKey(objName)) {
+        if (map.remove(objName) == null) {
```

Comment:
lost the CHECK here?

Reply:
I don't see a big reason to ignore detected problem here if `CHECK` is disabled. Before this change there was a reason because additional lookup was used for this check. But now it's only a comparison with `null`.